### ISSUE TITLE
TypeVar default refers to variables that are out of scope

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pyright#7553]
**Describe the bug**
Got that error while testing a PR to add TypeVar defaults to `slice`: https://github.com/python/typeshed/pull/11637.
The PEP suggests it to be defined that way so I'd assume this to be a false-positive but maybe I'm missing something.
https://peps.python.org/pep-0696/#using-another-type-parameter-as-default

```python
from typing import Generic, Self, overload
from typing_extensions import TypeVar

_StartT = TypeVar("_StartT", default=int)
_StopT = TypeVar("_StopT", default=_StartT)
_StepT = TypeVar("_StepT", default=int | None)

class slice(Generic[_StartT, _StopT, _StepT]):
    ...
    @property
    def stop(self) -> _StopT: ...  # error
    @overload
    def __new__(cls, stop: _StopT, /) -> Self: ...  # error
    @overload
    def __new__(cls, start: _StartT, stop: _StopT, step: _StepT = ..., /) -> Self: ...
    ...
```

```
  test.py:11:9 - error: Type parameter "_StopT" has a default type that refers to one or more type variables that are out of scope
    Type variable "_StartT" is not in scope (reportGeneralTypeIssues)
  test.py:13:9 - error: Type parameter "_StopT" has a default type that refers to one or more type variables that are out of scope
    Type variable "_StartT" is not in scope (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
pyright 1.1.355

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.356, which I just published.
